src/udiskslinuxdriveata.c

```diff
@@ -461,9 +480,12 @@ static gboolean get_pm_state (UDisksLinuxDevice *device, GError **error, guchar
       g_prefix_error (error, "Error sending ATA command CHECK POWER MODE: ");
       goto out;
     }
-  /* count field is used for the state, see ATA8: table 102 */
-  *count = output.count;
-  rc = TRUE;
+  /* state field is used for the state, see ATA8: table 102 */
+  *state = output.count;
+
+  if (state == 0xFF || state == 0x80)
+      rc = TRUE;
```

Comment:
In glib calling conventions the return value indicates whether the call was successful or not (and _error_ set when unsuccessful). Values retrieved by successful call should be returned some other way. It is considered an programming error to return _FALSE_ with no _error_ set when _error_ is non-NULL.